librz/core/cmd.c

```diff
@@ -4760,55 +4764,130 @@ DEFINE_HANDLE_TS_FCN_AND_SYMBOL(redirect_command) {
 	return res;
 }
 
+typedef struct _search_help {
+	bool color;
+	PJ *pj;
+} RzHelpSearch;
+
+static bool do_help_manual(RzCmd *cmd, const char *cmdname, bool color) {
+	if (!cmdname) {
+		return false;
+	}
+	char *cmdname_help = NULL;
+	char *help_msg = NULL;
+	RzCmdParsedArgs *help_pra = NULL;
+
+	cmdname_help = rz_str_newf("%s?", cmdname);
+	if (!cmdname_help) {
+		goto help_rec_err;
+	}
+	help_pra = rz_cmd_parsed_args_newcmd(cmdname_help);
+	if (!help_pra) {
+		goto help_rec_err;
+	}
+	ut32 pflags = color ? RZ_CMD_HELP_PRINT_USE_COLORS : 0;
+	help_msg = rz_cmd_get_help(cmd, help_pra, pflags);
```

Comment:
The logic you are using here is wrong because it just prints commands when you ask for the help of the parent command (for example, if you do `?* zsr` you won't see anything because nobody is going to print its summary. When you do `?* zs` you see `zsr` just because `zs` is printing its children).

I think you should make have a way to ask for just the summary line, like:
```
rz_cmd_get_help(cmd, help_pra, RZ_CMD_HELP_PRINT_SHOW_SUMMARY)
```
assuming help_pra.cmd == 'zsr`:
```
| zsr <newname>        # Rename selected zignspace
```